components/gameTerminal.js

```diff
@@ -0,0 +1,93 @@
+import styled from "styled-components";
+
+export default function GameTerminal({ previousMove, moveStatus }) {
+  console.log(previousMove);
+  console.log(moveStatus);
+
+  let piece;
+  switch (previousMove.piece) {
+    case "p":
+      piece = "Pawn";
+      break;
+    case "b":
+      piece = "Bishop";
+      break;
+    case "n":
+      piece = "Knight";
+      break;
+    case "r":
+      piece = "Rook";
+      break;
+    case "q":
+      piece = "Queen";
+      break;
+    case "k":
+      piece = "King";
+      break;
+    default:
+      piece = "piece";
+  }
+
+  let flag;
+  switch (previousMove.flags) {
+    case "b":
+      flag = "pushed two squares";
+      break;
+    case "e":
+      flag = "CAPTURED en passant";
+      break;
+    case "c":
+      flag = "CAPTURED";
+      break;
+    case "p":
+      flag = "PROMOTED";
+      break;
+    case "k":
+      flag = "castled kingside";
+      break;
+    case "q":
+      flag = "castled queenside";
+      break;
+    case "pc":
+      flag = "CAPTURED & PROMOTED";
+      break;
+    default:
+      flag = "moved";
+  }
+
+  let gameEnd;
+  if (moveStatus.isDraw) {
+    gameEnd = "Draw!";
+  } else if (moveStatus.isStalemate) {
+    gameEnd = "Stalemate!";
+  } else if (moveStatus.isThreefoldRep) {
+    gameEnd = "Threefold Repetition - Draw!";
+  } else {
+    gameEnd = "Checkmate!";
+  }
```

Comment:
This is quite repetitive, you can refactor the switch statements into a function like stated in a comment before, then you can create a "utils" folder or something, but these resolving Functions in there and import them where you need to resolve these Strings.

Reply:
you´re right about that. I just refactored the components, so that there is no repetition anymore. I´ll keep an eye out on that problem, if it reoccurs, I´ll move the functions to a separate folder.